### PR TITLE
Add spec for cherry-picker endpoint deployment

### PR DESCRIPTION
**Pull request by Rajalakshmi-Girish**
This PR is to add spec for `cherry-picker `endpoint deployment in IKS Prow .
cherry-pick bot is to created PRs on required release branch once master PR is merged for `ocp4-upi-power{vm|vs}` repos .

### CODE REVIEW

**Comment by ltccci** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/ppc64le-cloud/test-infra/pull/156#" title="Author self-approved">Rajalakshmi-Girish</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **mkumatag** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @mkumatag` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=ppc64le-cloud%2Ftest-infra).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/ppc64le-cloud/test-infra/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: config/prow/cluster/cherrypick_deployment.yaml
Comment (mkumatag) [inline config/prow/cluster/cherrypick_deployment.yaml:1]: List may not work in the k8s, it is actually a openshift constraint,.. lets devide these resources with seperator `---` like we did in the other files.

**Review by mkumatag — CHANGES_REQUESTED** [review]
looks good except one minor comment,

**Comment by ltccci** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/ppc64le-cloud/test-infra/pull/156#issuecomment-801146150" title="Approved">mkumatag</a>*, *<a href="https://github.com/ppc64le-cloud/test-infra/pull/156#" title="Author self-approved">Rajalakshmi-Girish</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=ppc64le-cloud%2Ftest-infra).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/ppc64le-cloud/test-infra/blob/master/OWNERS)~~ [mkumatag]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by Rajalakshmi-Girish** [comment]
@mkumatag 
I get the below error when I apply the spec:
```
root@sajauddin-dev:~/rajalakshmi/fork/test-infra# kubectl apply -f config/prow/cluster/cherrypick_deployment.yaml
service/cherrypick created
error: error validating "config/prow/cluster/cherrypick_deployment.yaml": error validating data: **[ValidationError(Deployment.spec.selector): unknown field "app" in io.k8s.apimachinery.pkg.apis.meta.v1.LabelSelector, ValidationError(Deployment.spec.selector): unknown field "component" in io.k8s.apimachinery.pkg.apis.meta.v1.LabelSelector];** if you choose to ignore these errors, turn validation off with --validate=false
root@sajauddin-dev:~/rajalakshmi/fork/test-infra#
```
As in other deployments, I think the labels have to be changed to
```
labels:
    app: cherrypick
```
**Instead of** 
```
labels:
    app: prow
    component: cherrypick

```

File: config/prow/cluster/cherrypick_deployment.yaml
Comment (mkumatag) [inline config/prow/cluster/cherrypick_deployment.yaml:29]: Problem is here.. need to fix the indentation for these items under matchLabels.. add 2 spaces and make these part of the matchLabels instead of selector..

**Review by mkumatag — CHANGES_REQUESTED** [review]

**Comment by ltccci** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/ppc64le-cloud/test-infra/pull/156#" title="Author self-approved">Rajalakshmi-Girish</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **mkumatag** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @mkumatag` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=ppc64le-cloud%2Ftest-infra).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/ppc64le-cloud/test-infra/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by ltccci** [comment]
New changes are detected. LGTM label has been removed.

File: config/prow/cluster/cherrypick_deployment.yaml
Comment (Rajalakshmi-Girish) [reply config/prow/cluster/cherrypick_deployment.yaml]: I fixed the indentation and also changed the value passed to `--github-token-path` flag from `/etc/github/oauth` to `/etc/github/token`. The `cherrypick pod` is Up and Running and below is the text from the pod log:
```
{"client":"github","file":"prow/github/client.go:705","func":"k8s.io/test-infra/prow/github.(*client).log","level":"info","msg":"User()","time":"2021-03-18T10:48:56Z"}
{"client":"github","file":"prow/github/client.go:705","func":"k8s.io/test-infra/prow/github.(*client).log","level":"info","msg":"User()","time":"2021-03-18T10:48:56Z"}
{"client":"github","file":"prow/github/client.go:705","func":"k8s.io/test-infra/prow/github.(*client).log","level":"info","msg":"GetRepos(ltccci, true)","time":"2021-03-18T10:48:56Z"}
{"client":"github","duration":"67.557213ms","file":"prow/github/client.go:707","func":"k8s.io/test-infra/prow/github.(*client).log.func2","level":"debug","msg":"GetRepos(ltccci, true) finished","time":"2021-03-18T10:48:56Z"}
```
@mkumatag Kindly confirm if the above message from the log means the cherrypicker Endpoint is working as expected. Please let me know if anything else needs to be verified.

Comment (mkumatag) [reply config/prow/cluster/cherrypick_deployment.yaml]: /lgtm

**Review by mkumatag — APPROVED** [review]
/approve

feel free to remove hold whenever you are comfortable

**Comment by ltccci** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/ppc64le-cloud/test-infra/pull/156#pullrequestreview-615307595" title="Approved">mkumatag</a>*, *<a href="https://github.com/ppc64le-cloud/test-infra/pull/156#" title="Author self-approved">Rajalakshmi-Girish</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=ppc64le-cloud%2Ftest-infra).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/ppc64le-cloud/test-infra/blob/master/OWNERS)~~ [mkumatag]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: config/prow/cluster/cherrypick_deployment.yaml
Comment (mkumatag) [inline config/prow/cluster/cherrypick_deployment.yaml:51]: Where is this used?

**Review by mkumatag — CHANGES_REQUESTED** [review]

**Comment by ltccci** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/ppc64le-cloud/test-infra/pull/156#" title="Author self-approved">Rajalakshmi-Girish</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **mkumatag** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @mkumatag` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=ppc64le-cloud%2Ftest-infra).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/ppc64le-cloud/test-infra/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: config/prow/cluster/cherrypick_deployment.yaml
Comment (Rajalakshmi-Girish) [reply config/prow/cluster/cherrypick_deployment.yaml:51]: I explicitly see this file not being used by the container.
I was thinking it might be used by the deployment at some stage as I do not know the internals of  `gcr.io/k8s-prow/cherrypicker `image.

Comment (mkumatag) [reply config/prow/cluster/cherrypick_deployment.yaml:51]: lets remove and see how it goes,

Comment (Rajalakshmi-Girish) [reply config/prow/cluster/cherrypick_deployment.yaml:51]: @mkumatag I get the below error when I remove `hmac` volume and corresponding mount:
```
{"error":"error reading /etc/webhook/hmac: open /etc/webhook/hmac: no such file or directory","file":"prow/external-plugins/cherrypicker/main.go:96","func":"main.main","level":"fatal","msg":"Error starting secrets agent.","time":"2021-03-21T12:08:42Z"}
```
Seems the code from below file needs the webhook secret details:
`https://github.com/kubernetes/test-infra/blob/master/prow/external-plugins/cherrypicker/main.go#L96`

Comment (mkumatag) [reply config/prow/cluster/cherrypick_deployment.yaml:51]: Okay..

**Review by mkumatag — APPROVED** [review]
/unhold

**Comment by ltccci** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/ppc64le-cloud/test-infra/pull/156#pullrequestreview-617015052" title="Approved">mkumatag</a>*, *<a href="https://github.com/ppc64le-cloud/test-infra/pull/156#" title="Author self-approved">Rajalakshmi-Girish</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=ppc64le-cloud%2Ftest-infra).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/ppc64le-cloud/test-infra/blob/master/OWNERS)~~ [mkumatag]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
